.github/workflows/main.yml

```diff
@@ -1,10 +1,39 @@
-on: [push]
+# @see https://github.com/EddieHubCommunity/awesome-github-profiles/issues/173
+name: Prettier Lint checker to find any syntax/spacing/best practices tweaks.
+
+# @see https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags
+# "Push" is every commit to any branch, whilst "pull_request" includes update commits to PRs.
+on: [push, pull_request]
 
 jobs:
+  prettier:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          # Make sure the actual branch is checked out when running on pull requests
+          ref: ${{ github.head_ref }}
+          # This is important to fetch the changes to the previous commit
+          fetch-depth: 0
+
+      - name: Prettify code
+        uses: creyD/prettier_action@v3.1
+        with:
+          # We only check ONLY_CHANGED as this is a PR action
+          only_changed: True
+          # We specify the prettier checker to not automatically overwrite the files.
+          dry: True
+          # This defaults to only "**/*.js", so we specify what we have as file extensions here.
+          prettier_options: --write **/*.{md,json,yml,yaml}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
   table:
     runs-on: ubuntu-latest
+    needs: prettier
     name: Update README from json data
-    if: github.repository_owner == 'EddieHubCommunity'
+    if: ${{ github.event === 'push' && github.repository_owner == 'EddieHubCommunity' }}
```

Comment:
Not sure if the extra check of `event === 'push'` is needed as the trigger is `push` and `pull_request`?

Reply:
We don't want to update the readme if it is a pull request though 😅